public/components/data_connections/components/access_control_tab.tsx

```diff
@@ -0,0 +1,205 @@
+  const [mode, setMode] = useState<'view' | 'edit'>('view');
+  const [roles, setRoles] = useState<Array<{ label: string }>>([]);
+  const [selectedQueryPermissionRoles, setSelectedQueryPermissionRoles] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [selectedAccelerationPermissionRoles, setSelectedAccelerationPermissionRoles] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [queryPermissionRadioSelected, setQueryRadioIdSelected] = useState(`query-1`);
+  const [accelerationPermissionRadioSelected, setAccelerationRadioIdSelected] = useState(
+    `acceleration-1`
+  );
+
+  const [finalQueryPermissionsPlaceHolder, setFinalQueryPermissionsPlaceHolder] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [
+    finalAccelerationPermissionsPlaceHolder,
+    setFinalAccelerationPermissionsPlaceHolder,
+  ] = useState<Array<{ label: string }>>([]);
+
+  useEffect(() => {
+    coreRefs.http!.get('/api/v1/configuration/roles').then((data) =>
+      setRoles(
+        Object.keys(data.data).map((key) => {
+          return { label: key };
+        })
+      )
+    );
+  }, []);
+
+  const queryRadios = [
+    {
+      id: QUERY_RESTRICT,
+      label: 'Restricted - accessible by users with specific OpenSearch roles',
+    },
+    {
+      id: QUERY_ALL,
+      label: 'Everyone - accessible by all users on this cluster',
+    },
+  ];
+  const accelerationRadios = [
+    {
+      id: ACCELERATION_RESTRICT,
+      label: 'Restricted - accessible by users with specific OpenSearch roles',
+    },
+    {
+      id: ACCELERATION_ALL,
+      label: 'Everyone - accessible by all users on this cluster',
+    },
+  ];
+
+  const renderViewAccessControlDetails = () => {
+    return (
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem grow={false}>
+              <EuiText className="overview-title">Query access</EuiText>
+              <EuiText size="s" className="overview-content">
+                {finalQueryPermissionsPlaceHolder.length
+                  ? `Restricted to ${finalQueryPermissionsPlaceHolder.map((role) => role.label)}`
+                  : '-'}
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem grow={false}>
+              <EuiText className="overview-title">Acceleration permissions</EuiText>
+              <EuiText size="s" className="overview-content">
+                {finalAccelerationPermissionsPlaceHolder.length
+                  ? `Restricted to ${finalAccelerationPermissionsPlaceHolder.map(
+                      (role) => role.label
+                    )}`
+                  : '-'}
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
+  const renderEditAccessControlDetails = () => {
```

Comment:
Please turn this into a formal React FC.
This is a quality issue for us throughout our plugin.

Reply:
I tried to address this in: [fc96304](https://github.com/opensearch-project/dashboards-observability/pull/992/commits/fc96304362b085696c2e5df8e7c638b69c142180), please let me know if I understood what you meant correctly.